python/lsst/analysis/tools/atools/metadataMetrics.py

```diff
@@ -0,0 +1,54 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("MetadataMetricTool",)
+
+from typing import Any, Dict, Iterable
+
+from lsst.pex.config import DictField, Field
+
+from ..interfaces import AnalysisAction, AnalysisTool
+
+
+class LoadDoubleKeyedData(AnalysisAction):
+    """Load double keyed data"""
+
+    name = Field[str](doc="The name of the key to load.")
+
+    def getInputSchema(self) -> Iterable[tuple[str, Dict]]:
```

Comment:
Why doesn't this return a `KeyedDataSchema`? That may necessitate a modification to `KeyedDataSchema`, but it seems in-keeping with the MO elsewhere in atools, right?

Reply:
Because it's not included [here](https://github.com/lsst/analysis_tools/blob/cdf17bb5f2d4739b48d7302d31eab345eaa8c960/python/lsst/analysis/tools/interfaces/_interfaces.py#L117):
```python
KeyedDataSchema = Iterable[tuple[str, type[Vector] | ScalarType | type[HealSparseMap] | type[Tensor]]]
```
I wasn't brave enough to modify `KeyedDataSchema`.

Reply:
Can you not squeeze your outputs into one of these types? I.e., you're currently returning a `Iterable[tuple[str, dict]]` - why does this need to be a `dict`? All you're really interested in is the amp offset value, right? Your concern is that it might make it difficult to interpret across cameras?

Reply:
Now updated to:
```python
    def getInputSchema(self) -> Iterable[tuple[str, Mapping[str, Any]]]:
        return [(self.name, Mapping[str, Any])]
```

> Can you not squeeze your outputs into one of these types? I.e., you're currently returning a `Iterable[tuple[str, dict]]` - why does this need to be a `dict`? All you're really interested in is the amp offset value, right?

I can, but then we reach a point where the pipeline yaml needs to be camera-specific, providing explicit information about the metadata for the specific camera we're interested in — not ideal. Furthermore, to squeeze our outputs into one of those types in `KeyedDataSchema`, we'll need to modify the `runQuantum` method in `IsrMetadataExposureDetectorAnalysisTask`. However, making these changes would narrow the task down to the `ampOffset` subtask and defeat the purpose of keeping it versatile for all types of ISR metadata.

> Your concern is that it might make it difficult to interpret across cameras?

Correct.

Reply:
Following your suggestion, I slightly modified `KeyedData`, `KeyedDataSchema`, and `KeyedDataTypes` to include the `Mapping` type and stay aligned with the MO elsewhere in atools.